backend/src/util/tokenChecker.js

```diff
@@ -0,0 +1,20 @@
+const JWT = require("jsonwebtoken");
+
+const tokenChecker = (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  if (authHeader) {
+    const token = authHeader.split(' ')[1];
+    JWT.verify(token, 'SECRET', (err, decoded) => {
+      if (err) {
+        return res.status(401).json({ error: 'Token is not valid' });
+      } else {
+        req.decoded = decoded;
+        next();
```

Comment:
next(); sends the request along to the next middleware or in our case the 
gameRouter